src/Map/src/Bridge/Leaflet/README.md

```diff
@@ -37,6 +37,62 @@ $leafletOptions = (new LeafletOptions())
 $map->options($leafletOptions);
 ```
 
+## Use cases
+
+Below are some common or advanced use cases when using a map.
+
+> **Note**:
+> As the UX Map component is new, there are not too many examples for the moment.
+
+### Customize the marker
+
+A common use case is to customize the marker. You can listen to the `ux:map:marker:before-create` event to customize the marker before it is created.
+
+Assuming you have a map with a custom controller:
+```twig
+{{ render_map(map, {'data-controller': 'my-map' }) }}
+```
+
+You can create a controller to customize the marker before it is created:
+```js
+// assets/controllers/my_map_controller.js
+import {Controller} from "@hotwired/stimulus";
+
+export default class extends Controller
+{
+    connect() {
+        this.element.addEventListener('ux:map:marker:before-create', this._onMarkerBeforeCreate);
+    }
+
+    disconnect() {
+        // Always remove listeners when the controller is disconnected
+        this.element.removeEventListener('ux:map:marker:before-create', this._onMarkerBeforeCreate);
+    }
+
+    _onMarkerBeforeCreate(event) {
+        // You can access the marker definition and the Leaflet object
+        // Note: `definition.rawOptions` is the raw options object that will be passed to the `L.marker` constructor. 
+        const { definition, L } = event.detail;
+
+        // Use a custom icon for the marker
+        const redIcon = L.icon({
+          // Note: instead of using an hardcoded URL, you can use the `extra` parameter from `new Marker()` (PHP) and access it here with `definition.extra`.
+          iconUrl: 'https://leafletjs.com/examples/custom-icons/leaf-red.png',
+          shadowUrl: 'https://leafletjs.com/examples/custom-icons/leaf-shadow.png',
+          iconSize: [38, 95], // size of the icon
+          shadowSize: [50, 64], // size of the shadow
+          iconAnchor: [22, 94], // point of the icon which will correspond to marker's location
+          shadowAnchor: [4, 62],  // the same for the shadow
+          popupAnchor: [-3, -76] // point from which the popup should open relative to the iconAnchor
+        })
+  
+        definition.rawOptions = {
+          icon: redIcon,
+        }
+    }
```

Comment:
This will update all maps on the page (if multiple ones) right ?

Reply:
Nope, only the ones using `my-map` as `data-controller`